api/routes/user.py

```diff
@@ -20,39 +28,83 @@ def create_user(req_user: UserCreate, db: Session = default_session):
     return user.create(db, req_user)
 
 
-@router.get("/", response_model=List[UserInfo])
+@router.get("/", response_model=List[UserInfoAllAllowedFields])
 def get_all_users(
-    offset: int, limit: int, db: Session = default_session
+    offset: int,
+    limit: int,
+    db: Session = default_session,
+    Auth: AuthJWT = default_authJWT,
 ):
     """
     Get all the Users stored in database
     """
+    auth.is_only_admin_permitted(db, Auth)
+
     return user.get_all(db, offset, limit)
 
 
-@router.get("/{user_id}", response_model=UserInfo)
-def get_user(user_id: int, db: Session = default_session):
+@router.get(
+    "/{user_id}",
+    response_model=Union[UserInfoAllAllowedFields, UserInfo],
+)
+def get_user(
+    user_id: int,
+    db: Session = default_session,
+    Auth: AuthJWT = default_authJWT,
+):
     """
     Get the User with the given ID
     """
-    return user.get_by_id(db, user_id)
+    current_user = auth.get_authenteficated_user(db, Auth)
+    db_user = user.get_by_id(db, user_id)
+    if current_user.is_admin:
+        return UserInfoAllAllowedFields(**db_user.__dict__)
+
+    return UserInfo(**db_user.__dict__)
 
 
-@router.put("/{user_id}", response_model=UserInfo)
+@router.put(
+    "/{user_id}",
+    response_model=Union[UserInfoAllAllowedFields, UserInfo],
+)
 def update_user(
-    user_id: int, req_user: UserUpdate, db: Session = default_session
+    user_id: int,
+    req_user: UserUpdateAllAllowedFields,
+    db: Session = default_session,
+    Auth: AuthJWT = default_authJWT,
 ):
     """
     Update an User stored in the database
     """
-    return user.update(db, user_id, req_user)
+    current_user = auth.get_authenteficated_user(db, Auth)
+    if not current_user.is_admin:
+        if current_user.id != user_id:
+            raise HTTPException(status_code=status.HTTP_403_FORBIDDEN)
+        if isinstance(req_user.is_active, bool):
+            raise HTTPException(
+                status_code=status.HTTP_403_FORBIDDEN,
+                detail="Update status not allowed",
+            )
+
+        req_user = UserUpdate(**req_user.__dict__)
+
+    updated_user = user.update(db, user_id, req_user)
+    if current_user.is_admin:
+        return UserInfoAllAllowedFields(**updated_user.__dict__)
+
+    return UserInfo(**updated_user.__dict__)
 
 
 @router.delete("/{user_id}")
-def delete_user(user_id: int, db: Session = default_session):
+def delete_user(
+    user_id: int,
+    db: Session = default_session,
+    Auth: AuthJWT = default_authJWT,
+):
     """
     Delete the User with the given ID
     """
+    auth.is_only_admin_permitted(db, Auth)
     user.delete(db, user_id)
 
     return "User deleted successfully"
```

Comment:
You can use smth like this for cases where you return only status message:
```
class StatusResponse(BaseResponse):
    success: bool
    message: Optional[str] = None
```

Reply:
I added it